packages/protocol/src/access/TimeLock.sol

```diff
@@ -0,0 +1,330 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.15;
+
+/**
+ * @title TimeLock.
+ * @author Fujidao Labs
+ * @notice Enforce a time lock in Fuji-V2 risk parameter change operations. This contract is
+ * based on OpenZeppelin-TimelockController; however, all access control calls
+ * are delegated to {Chief}.
+ * Changes are indicated through documentation
+ * `SupportsInterface` and `onReceived` token functions were removed from the OZ implementation.
+ */
+
+import {SystemAccessControl} from "./SystemAccessControl.sol";
+
+contract TimeLock is SystemAccessControl {
```

Comment:
I partially see the point of having our own implementation but why not inherit from OZ's "TimelockController" and only have the chief set in the constructor?

btw OZ use "Timelock" and not "TimeLock"

Reply:
No, actually, I don't get the need to have our own implementation. On the contrary, this will prevent us from using the tools for managing timelocks like Defender https://defender.openzeppelin.com/#/admin/addTimelock

Reply:
Under the assumption we wanted to handle all access control authorization at a central hierarchy in the Chief contract, the Openzeppelin timelock implements its own separate AccessControl. It defines its roles within it. 

That would mean that roles like Default_ADMIN within Chief, have no control whatsoever on the Timelock. 

I propose we first experiment with Defender timelock on a testnest and see what are the "special" features about it. Otherwise, we commit ourselves that when our team scales, we will have to rely more and more on the paid-version of it.

Reply:
Renamed TimeLock to Timelock